File: 3-recognize-valid/input.c
#include <stdio.h>

int main() {
    int positiveInt = 12;
    int negativeInt = -12;
    double positiveFrac = 1.5;
    double negativeFrac = -1.5;
    printf("hello world");
    
    char x[] = "hello world";
    int arr[] = {1,2,3};
    
    for (int i = 0; i < 3; i++) {
        printf("%d\n", arr[i]);
    }
    
    return 0;
}

void greet() {
    printf("greetings!");
}

struct Person {
    char name[50];
    int age;
};

int add(int a, int b) {
    if (a+b > 10){
        return a + b;
    }else {
        return -1;
    }
}

File: 3-recognize-valid/input.java
public class HelloWorld {
    private String name;

    public static void main(String[] args) {

            for (int i = 0; i < 10; i++) {
                System.out.println(i);
            }

            int a = 'a';
            int b = 15;
            float c = -5.5;
            double d = 5.6;
            boolean truth = 1 < 2 && 4 > 5 || 8==8;
            System.out.println("hello world\n with new line");
            
    }

    public void greet() {
        System.out.println("greetings!");
    }

    public void sayName() {
        System.out.println("name is " + name);
    }
}

//comment
/* multiline 
 * comment
*/

interface Animal {
    void makeSound();
}

class Dog implements Animal {
    public void makeSound(boolean loud) {
        if (loud){
            System.out.println("WOOF");
        }else{
            System.out.println("woof");
        }
    }
}
